src/Html/HtmlCssSelectors.fs

```diff
@@ -0,0 +1,375 @@
+                    tokenize' (AttributeName(getOffset t + 1, s) :: OpenAttribute(getOffset t) :: acc) t'
+                | ']' :: t ->
+                    tokenize' (CloseAttribute(getOffset t) :: acc) t
+                | '=' :: t ->
+                    let s, t' = readString "" t
+                    tokenize' (AttributeValue(getOffset t + 1, s) :: Assign(getOffset t) :: acc) t'
+                | '$' :: '=' :: t ->
+                    let s, t' = readString "" t
+                    tokenize' (AttributeValue(getOffset t + 1, s) :: EndWith(getOffset t) :: acc) t'
+                | '^' :: '=' :: t ->
+                    let s, t' = readString "" t
+                    tokenize' (AttributeValue(getOffset t + 1, s) :: StartWith(getOffset t) :: acc) t'
+                | '|' :: '=' :: t ->
+                    let s, t' = readString "" t
+                    tokenize' (AttributeValue(getOffset t + 1, s) :: AttributeContainsPrefix(getOffset t) :: acc) t'
+                | '*' :: '=' :: t ->
+                    let s, t' = readString "" t
+                    tokenize' (AttributeValue(getOffset t + 1, s) :: AttributeContains(getOffset t) :: acc) t'
+                | '~' :: '=' :: t ->
+                    let s, t' = readString "" t
+                    tokenize' (AttributeValue(getOffset t + 1, s) :: AttributeContainsWord(getOffset t) :: acc) t'
+                | '!' :: '=' :: t ->
+                    let s, t' = readString "" t
+                    tokenize' (AttributeValue(getOffset t + 1, s) :: AttributeNotEqual(getOffset t) :: acc) t'
+                | StartsWith ":checkbox" t  ->
+                    tokenize' (Checkbox(getOffset t + 1) :: acc) t
+                | StartsWith ":selected" t  ->
+                    tokenize' (Selected(getOffset t + 1) :: acc) t
+                | StartsWith ":checked" t ->
+                    tokenize' (Checked(getOffset t + 1) :: acc) t
+                | StartsWith ":button" t -> 
+                    tokenize' (Button(getOffset t + 1) :: acc) t
+                | StartsWith ":hidden" t -> 
+                    tokenize' (Hidden(getOffset t + 1) :: acc) t
+                | StartsWith ":radio" t -> 
+                    tokenize' (Radio(getOffset t + 1) :: acc) t
+                | StartsWith ":password" t -> 
+                    tokenize' (Password(getOffset t + 1) :: acc) t
+                | StartsWith ":empty" t ->
+                    tokenize' (EmptyNode(getOffset t + 1) :: acc) t
+                | TokenStr ":disabled" t ->
+                    let _, t' = readString "" t
+                    tokenize' (Disabled(getOffset t + 1) :: acc) t'
+                | StartsWith ":enabled" t ->
+                    tokenize' (Enabled(getOffset t + 1) :: acc) t
+                | StartsWith ":file" t ->
+                    tokenize' (File(getOffset t + 1) :: acc) t
+                | '>' :: t ->
+                    let seqtoken = acc |> List.toSeq |> Seq.skip(1) |> Seq.toList
```

Comment:
Good job! This is the future in order to replace htmltable and a more
flexible way to parse Html!
On Oct 21, 2015 9:29 AM, "Jérémie Chassaing" <notifications@github.com>
wrote:

> In src/Html/HtmlCssSelectors.fs
> <https://github.com/fsharp/FSharp.Data/pull/886#discussion_r42591118>:
>
> > +                    tokenize' (Hidden(getOffset t + 1) :: acc) t
> > +                | StartsWith ":radio" t ->
> > +                    tokenize' (Radio(getOffset t + 1) :: acc) t
> > +                | StartsWith ":password" t ->
> > +                    tokenize' (Password(getOffset t + 1) :: acc) t
> > +                | StartsWith ":empty" t ->
> > +                    tokenize' (EmptyNode(getOffset t + 1) :: acc) t
> > +                | TokenStr ":disabled" t ->
> > +                    let _, t' = readString "" t
> > +                    tokenize' (Disabled(getOffset t + 1) :: acc) t'
> > +                | StartsWith ":enabled" t ->
> > +                    tokenize' (Enabled(getOffset t + 1) :: acc) t
> > +                | StartsWith ":file" t ->
> > +                    tokenize' (File(getOffset t + 1) :: acc) t
> > +                | '>' :: t ->
> > +                    let seqtoken = acc |> List.toSeq |> Seq.skip(1) |> Seq.toList
>
> You can use acc |> List.tail directly here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fsharp/FSharp.Data/pull/886/files#r42591118>.
>